### PR TITLE
allows user defined ingress annotations

### PR DESCRIPTION
**Pull request by szferi**
Extend the ingress definitions with user-defined annotations so she can do something like this:
```
ingress: {
	provider: "github.com/ns-test/cmingress"
	domains: ["example.com", "www.example.com"]
	annotations: {
		"cert-manager.io/cluster-issuer": "cert-issuer-staging"
	}
}
```

### CODE REVIEW

File: internal/frontend/cuefrontendopaque/service.go [hunk 1]
```diff
@@ -46,10 +46,12 @@ type cueIngress struct {
...
 type CueIngressDetails struct {
 	// Key is domain.
-	HttpRoutes     map[string][]string `json:"httpRoutes"`
-	ProviderClass  string              `json:"provider"`
-	Domains        []string            `json:"domains"`
-	AllowedOrigins []string            `json:"allowed_origins"`
+	HttpRoutes    map[string][]string `json:"httpRoutes"`
+	ProviderClass string              `json:"provider"`
+	Domains       []string            `json:"domains"`
+	// Certificates   map[string]secretRef   `json:"certificates,omitempty"`
```
Comment (hugosantos) [inline internal/frontend/cuefrontendopaque/service.go:52]: Nit: please remove.

File: schema/server.proto [hunk 2]
```diff
@@ -102,8 +104,12 @@ message Server {
         Endpoint.Type            endpoint_type    = 7;
         PackageRef               ingress_provider = 8;
         repeated DomainSpec      ingress_domain   = 10;  // Only has a meaning if this is an ingress.
-
-        reserved 2;
+        repeated IngressAnnotation ingress_annotation = 11;
```
Comment (hugosantos) [inline schema/server.proto:107]: Would it make sense to use "ServiceAnnotations" here as well?

**Review by hugosantos — COMMENTED** [review]
Looks good! Just one question.

File: schema/server.proto (hunk 2 above)
Comment (szferi) [reply schema/server.proto:107]: Yes I was unsure about this but certainly we can

**Review by hugosantos — APPROVED** [review]

**Pull request merged**